source/Ship.cpp

```diff
@@ -3138,12 +3170,23 @@ int Ship::BaysTotal(const string &category) const
 {
 	int count = 0;
 	for(const Bay &bay : bays)
-		count += (bay.category == category);
+	{
+		if(bay.CanContain(category))
+			++count;
+	}
 	return count;
 }
 
 
 
+// Get the types of bays that this ship has and the number of each.
+map<string, int> Ship::BayTypeCounts() const
```

Comment:
now it would make sense to return a const&
```suggestion
const map<string, int> &Ship::BayTypeCounts() const
```